src/components/Charts/PriceChart/utils.ts

```diff
@@ -1,5 +1,101 @@
-import { timeDay, timeHour, TimeInterval, timeMinute, timeMonth } from 'd3'
-import { TimePeriod } from 'graphql/data/util'
+import { bisector, ScaleLinear, timeDay, timeHour, TimeInterval, timeMinute, timeMonth } from 'd3'
+import { PricePoint, TimePeriod } from 'graphql/data/util'
+
+/**
+ * Returns the minimum and maximum values in the given array of PricePoints.
+ */
+export function getPriceBounds(prices: PricePoint[]): { min: number; max: number } {
+  if (!prices.length) return { min: 0, max: 0 }
+
+  let min = prices[0].value
+  let max = prices[0].value
+
+  for (const pricePoint of prices) {
+    if (pricePoint.value < min) {
+      min = pricePoint.value
+    }
+    if (pricePoint.value > max) {
+      max = pricePoint.value
+    }
+  }
+
+  return { min, max }
+}
+
+/**
+ * Cleans up an array of PricePoints by removing zero values and marking gaps in data as blanks.
+ *
+ * @param prices - The original array of PricePoints
+ * @returns An object containing two arrays: fixedChart and blanks
+ */
+export function cleanUpPricePoints(prices: PricePoint[]) {
```

Comment:
can we just call this `cleanPricePoints` ? i feel like "clean up" is almost slang